sql/moz-fx-data-shared-prod/telemetry_derived/clients_daily_v6/metadata.yaml

```diff
@@ -1,21 +1,34 @@
----
-friendly_name: Clients Daily
-description: |-
-  A daily aggregation over `main` pings from each Firefox
+bigquery:
+  cluster_by:
+  - normalized_channel
+  - sample_id
+  partition_by:
+    field: submission_date
+    require_partition_filter: true
+    type: day
```

Comment:
In order to get the schema of a deployed table for validation via a dry run the partitioning fields need to be known. So I think having this information as part of the metadata might be useful. 

This information is also required if a new table is created based on a schema. Since the schema doesn't include this information we'll need to specify it here.

Reply:
I'm very much in favor of codifying partioning/clustering info here. I'd like us to mimic the BQ API representation in the metadata as much as possible, which I expect will make it more natural to be able to evolve this to take advantage of new features in the future. So I would expect this to look like:

```
bigquery:
  require_partition_filter: true
  clustering:
    fields:
    - normalized_channel
    - sample_id
  time_partitioning:
    field: submission_date
    require_partition_filter: true
    type: day
 ```